plugins/inputs/http/http_test.go

```diff
@@ -446,3 +451,74 @@ func TestHTTPWithCSVFormat(t *testing.T) {
 	require.NoError(t, acc.GatherError(plugin.Gather))
 	testutil.RequireMetricsEqual(t, expected, acc.GetTelegrafMetrics(), testutil.IgnoreTime())
 }
+
+const (
+	httpOverUnixScheme = "http+unix"
+)
+
+func TestConnectionOverUnixSocket(t *testing.T) {
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/data" {
+			w.Header().Set("Content-Type", "text/csv")
+			_, _ = w.Write([]byte(simpleCSVWithHeader))
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+
+	unixListenAddr := filepath.Join(os.TempDir(), fmt.Sprintf("httptestserver.%d.sock", rand.Intn(1_000_000)))
+	t.Cleanup(func() { os.Remove(unixListenAddr) })
+
+	unixListener, err := net.Listen("unix", unixListenAddr)
+	require.NoError(t, err)
```

Comment:
How about using [`CreateTemp`](https://pkg.go.dev/os#CreateTemp) here?
```suggestion
	tmpFile, err := os.CreateTemp("", "httptestserver.*.sock")
	require.NoError(t, err)
	defer os.Remove(tmpFile.Name())
	
	unixListenAddr := tmpFile.Name()
	require.NoError(t, tmpFile.Close()) 

	unixListener, err := net.Listen("unix", unixListenAddr)
	require.NoError(t, err)
```

Reply:
`net.Listen` _requires_ that the socket path does **not** already exist or else its gonna throw the following error:
```
bind: address already in use
```

Reply:
Understood. Thanks for the clarification!